she_codes_news/news/views.py

```diff
@@ -1,16 +1,35 @@
 from django.views import generic
+from django.urls import reverse_lazy
 from .models import NewsStory
+from .forms import StoryForm
 
 
 class IndexView(generic.ListView):
     template_name = 'news/index.html'
 
     def get_queryset(self):
         '''Return all news stories.'''
-        return NewsStory.objects.all()
+        return NewsStory.objects.all().order_by('-pub_date')
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
-        context['latest_stories'] = NewsStory.objects.all()[:4]
-        context['all_stories'] = NewsStory.objects.all()
-        return context
+        context['all_stories'] = self.get_queryset()
```

Comment:
could have used `context_object_name = 'all_stroies'` to avoid having to include this line, but it still does the same job.